python/src/lib.rs

```diff
@@ -94,9 +95,10 @@ enum PartitionFilterValue<'a> {
 
 #[pyclass]
 struct RawDeltaTable {
-    _table: deltalake::DeltaTable,
+    _table: Arc<Mutex<deltalake::DeltaTable>>,
     // storing the config additionally on the table helps us make pickling work.
     _config: FsConfig,
+    _rt: tokio::runtime::Runtime,
```

Comment:
Since we are now tracking a runtime directly on the table, we should likely figure how how we can share that best with both the filesystem implementation and the data checker.

Reply:
Actually should we have a global runtime for the Python bindings? IiUC if each table has a separate runtime then each one has their own thread pool which probably isn’t a great idea.

Reply:
Agreed - this would ideal. Had a quick read through the pyo3-asyncio docs in the hopes that they would somehow have a way to manage a runtime, but this does not seem to be the case. I guess this would bring us to a session of some sorts, or at least some globally managed state?

There are very likely other solutions, but off the top of my head we may be able to create some sort of python singleton, that houses the runtime, which we can pass around transparent to the user.

Reply:
Agreed! I will try to find a way to integrate a Python singleton and let you know!